sdk/storage/Azure.Storage.Common/swagger/Generator/src/generator.ts

```diff
@@ -913,7 +938,78 @@ function generateObject(w: IndentWriter, model: IServiceModel, type: IObjectType
+                        const separator = IndentWriter.createFenceposter();
+                        for (const property of properties) {
+                            if (separator()) { w.line(`,`); }
+                            w.write(`${types.getDeclarationType(property.model, property.required, property.readonly)} ${naming.parameter(property.clientName)}`);
+                        }
+                        w.line(`)`);
+                        w.scope('{', '}', () => {
+                            for (const property of properties) {
+                                w.line(`${naming.property(property.clientName)} = ${naming.parameter(property.clientName)};`);
+                            }
+                        });
+                    });
+                    w.line();
+                    w.line(`/// <summary>`)
+                    w.line(`/// Check if two ${naming.type(type.name)} instances are equal.`);
+                    w.line(`/// </summary>`);
+                    w.line(`/// <param name="other">The instance to compare to.</param>`);
+                    w.line(`/// <returns>True if they're equal, false otherwise.</returns>`);
+                    w.line(`public bool Equals(${naming.type(type.name)} other)`);
+                    w.scope('{', '}', () => {
+                        for (const property of properties) {
+                            const a = naming.property(property.clientName);
+                            const b = `other.${naming.property(property.clientName)}`;
+                            if (types.getDeclarationType(property.model, property.required, property.readonly) === "string") {
+                                w.line(`if (!System.StringComparer.Ordinal.Equals(${a}, ${b}))`);
+                            } else if (types.getDeclarationType(property.model, property.required, property.readonly).includes("[]")) {
+                                w.line(`if (!System.Collections.StructuralComparisons.StructuralEqualityComparer.Equals(${a}, ${b}))`);
+                            } else {
+                                w.line(`if (!${a}.Equals(${b}))`);
+                            }
+                            w.scope('{', '}', () => {
+                                w.line(`return false;`);
+                            });
+                        }
+                        w.line();
+                        w.line(`return true;`);
+                    });
+                    w.line();
+                    w.line(`/// <summary>`);
+                    w.line(`/// Check if two ${naming.type(type.name)} instances are equal.`);
+                    w.line(`/// </summary>`);
+                    w.line(`/// <param name="obj">The instance to compare to.</param>`);
+                    w.line(`/// <returns>True if they're equal, false otherwise.</returns>`);
+                    w.line(`public override bool Equals(object obj) => obj is ${naming.type(type.name)} && Equals((${naming.type(type.name)})obj);`);
```

Comment:
@pakrym - should we add an `[EditorBrowsable(false)]` here?

Reply:
Yep